javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -73,71 +76,91 @@ private static int treeSize(int branchCount, int depthShift) {
         }
     }
 
-    BitMappedTrie<T> prepend(T leading) {
-        final int newSize = length() + 1;
-        if (length() == 0) {
-            return new BitMappedTrie<>(asArray(leading), offset, newSize, depthShift);
-        } else {
-            Object[] array = this.array;
-            int shift = depthShift, offset = this.offset;
-            if (isFullLeft()) {
+    BitMappedTrie<T> prependAll(Iterable<? extends T> iterable) {
+        BitMappedTrie<T> result = this;
+        for (java.util.Iterator<? extends T> iterator = reverseIterator(iterable); iterator.hasNext(); ) {
+            Object[] array = result.array;
+            int shift = result.depthShift, offset = result.offset;
+            if (result.isFullLeft()) {
                 array = copyUpdate(Arrays.empty(), BRANCHING_FACTOR - 1, array);
                 shift += BRANCHING_BASE;
                 offset = treeSize(BRANCHING_FACTOR - 1, shift);
             }
 
-            offset -= 1;
-            array = modifyLeaf(array, shift, offset, COPY_NODE, updateLeafWith(leading));
-            return new BitMappedTrie<>(array, offset, newSize, shift);
+            final MutableInt delta = new MutableInt(0);
```

Comment:
`MutableInt` is more elegant than a single element `int[]`.
We need this for multiple return (without using `Tuples` and boxing)